src/components/bootcamp-interns/BootcampInternGrid.tsx

```diff
@@ -0,0 +1,156 @@
+import DeleteIcon from '@mui/icons-material/Delete'
+import { DataGrid, GridColumns, GridRenderCellParams, GridSelectionModel } from '@mui/x-data-grid'
+
+import { BootcampIntern } from 'lib/interfaces/BootcampIntern'
+import { useBootcampInternsList } from 'common/hooks/bootcampIntern'
+
+import DeleteModal from './DeleteModal'
+import { drawerWidth } from './MyDrawer'
+import BootcampModal from './BootcampModal'
+import DataGridHeader from './DataGridHeader'
+
+const useStyles = makeStyles(() => {
+  return {
+    internDataGrid: {
+      marginLeft: drawerWidth,
+      backgroundColor: '#f9f9f9',
+      width: '100%',
+      border: 'none',
+      borderTop: '1px solid lightgray',
+    },
+    dataGridHeader: {
+      marginLeft: drawerWidth,
+      marginTop: '40px',
+    },
+    icons: {
+      cursor: 'pointer',
+    },
+    datagridParent: {
+      display: 'flex',
+      width: '895px',
+      height: 'auto',
+      alignItems: 'center',
+      flexFlow: 'column wrap',
+      justifyContent: 'center',
+      marginLeft: drawerWidth,
+    },
+  }
+})
+
+const iconsStyle = { cursor: 'pointer' }
+
+export default function BootcampInternGrid() {
+  const [open, setOpen] = useState<boolean>(false)
+  const [selectedRows, setSelectedRows] = useState<GridSelectionModel>([])
+  const [details, setDetails] = useState<null | string[]>(null)
+  const [deleteData, setDeleteData] = useState<string | unknown>('')
+  const [deleteOpen, setDeleteOpen] = useState<boolean>(false)
+
+  const router = useRouter()
+  const classes = useStyles()
+
+  const { data }: UseBaseQueryResult<BootcampIntern[]> = useBootcampInternsList()
+
+  const columns: GridColumns = [
+    { field: 'id', headerName: 'ID', hide: true },
+    {
+      field: 'firstName',
+      headerName: 'First name',
+      width: 230,
+      align: 'center',
+      headerAlign: 'center',
+    },
+    {
+      field: 'lastName',
+      headerName: 'Last name',
+      width: 230,
+      align: 'center',
+      headerAlign: 'center',
+    },
+    {
+      field: 'email',
+      headerName: 'Email',
+      width: 230,
+      align: 'center',
+      headerAlign: 'center',
+    },
+    {
+      field: 'Actions',
+      headerName: 'Actions',
+      renderCell: (cellValues) => {
+        return (
+          <ButtonGroup>
+            <InfoIcon
+              color="success"
+              onClick={() => detailsClickHandler(cellValues)}
+              style={iconsStyle}
+            />
+            <EditIcon onClick={() => editClickHandler(cellValues)} style={iconsStyle} />
+            <DeleteIcon
+              color="error"
+              onClick={() => deleteClickHandler(cellValues)}
+              style={iconsStyle}
+            />
+          </ButtonGroup>
+        )
+      },
+      width: 150,
+      align: 'right',
+      headerAlign: 'center',
+    },
+  ]
+
+  const editClickHandler = (cellValues: GridRenderCellParams) => {
+    router.push(`/bootcamp-interns/${cellValues.id}/edit`)
```

Comment:
These internal urls are better to be stored in a single location again and that's `routes.ts` file with `string` or `() => string` definitions.

This will improve code clarity and will help with refactoring later on